src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,71 @@
+import cx from "classnames";
+import React, { FC, ReactElement, useEffect, useState } from "react";
+import { SystemTheme, Theme, ThemeColor } from "./ThemeProviderConstants";
+import { generateRandomAlphaString, generateThemeCssOverride, shouldGenerateTheme } from "./ThemeProviderUtils";
+import { withStaticProps } from "../../types";
+
+export interface ThemeProviderProps {
+  /**
+   * The theme to apply, consists of name - name of css class that will be added to the children, which should be unique - and the object of colors overrides for each system theme.
+   */
+  theme?: Theme;
+  /**
+   * The children to render with the theme
+   */
+  children: ReactElement;
+  /**
+   * Random string which adds up to theme name to make it more specific (in case if theme.name is colliding with some other class name)
+   */
+  randomStringSelector?: string;
```

Comment:
Do we want this prop, @shlomitc?
Apart from the tests, I guess I can imagine cases where it might be useful, though don't really like to garbage ThemeProviderProps with it... WDYT?

Reply:
1. it's not random, so the name might be misleading
2. if not mandatory, let's start without it and add it only when needed

Reply:
@orrgottlieb wanted to add this to be more secure in cases where e.g. when theme.name is the same as some other class name - so it would still work
Let's rename it to `additionalStringSelector`? wdyt @shlomitc ?

Reply:
1. `additionalStringSelector` still sounds a bit vague, maybe `specifyThemeClass` or something that can explain it without looking at the prop description
2. if we already provide the way to control it via `theme.name`, why do we need an additional prop for that?

Reply:
1. Renamed it to `themeClassSpecifier`
2. `theme.name` might not be unique and collide with other classes leading to issues... This additional `specifier` adds by default is random string, which makes sure that css selector is unique. We are allowing to provide it as a prop, so it won't hurt e.g. tests and this kind of thing - to grant consistency, if it's needed. for any reason